app/controllers/order_items_controller.rb

```diff
@@ -0,0 +1,82 @@
+class OrderItemsController < ApplicationController
+  before_action :check_if_product_is_in_cart, only: [:create]
+
+  def create
+    @order = current_order
+    @order_item = @order.order_items.create(order_item_params)
+    if @order_item.save
+      redirect_to cart_path
+    else
+      if @order_item.product.inventory == 1
+        flash[:error] = "Sorry, there is only #{@order_item.product.inventory} #{@order_item.product.name} available."
+      elsif @order_item.product.inventory == 0
+        flash[:error] = "Sorry, there are no #{@order_item.product.name.pluralize} available."
+      else
+        flash[:error] = "Sorry, there are only #{@order_item.product.inventory} #{@order_item.product.name.pluralize} available."
```

Comment:
Nice attention to detail here with an option and check to pluralize item(s). That is a great way of thinking about the user experience and how display messages should be viewed.